server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -582,10 +626,59 @@ public void uploadMetadata(
             );
             try {
                 try (IndexOutput indexOutput = storeDirectory.createOutput(metadataFilename, IOContext.DEFAULT)) {
+                    // TODO: The following regex could work incorrectly if both major and minor versions are double-digits.
+                    // This is because the major and minor versions do not have a separator in the filename currently
+                    // (Lucence<major><minor>).
+                    // We may need to revisit this if the filename pattern is updated in future Lucene versions.
+                    Pattern pattern = Pattern.compile("_\\d+_\\d+_Lucene(\\d+)\\d+_\\d+");
```

Comment:
I am not completely sure about using RegEx. Doesn't `Lucene` provide any utility around these filenames?
Also, we can abstract out logic to get version out of filename in a util. Check `RemoteStoreUtils` class.

Reply:
Ack, will move this to `RemoteStoreUtils` that has been introduced. 

`Lucene` does offer helper methods to parse dot-notation versions or `LUCENE_XY`/`LUCENE_X_Y`/`LUCENE_X_Y_Z` formatted versions ([docs](https://lucene.apache.org/core/9_0_0/core/org/apache/lucene/util/Version.html#:~:text=to%20use%20%22-,LUCENE_X_Y,-%22%20constant%20names)). But, in our case, the filename is coming as `LuceneXY` (i.e. the `_` between `Lucene` and `XY` is missing). Adding `_` will again involve some form of string parsing. But doing that and then using `parseLeniently()` is an alternate approach we can look at. However, the `parseLeniently()` docs note that `This API is for internal purposes only and might change in incompatible ways in the next release`.

Neither approach is as robust as we'd like. What are your thoughts on this?